src/library_gl.js

```diff
@@ -33,14 +33,14 @@ var LibraryGL = {
     currentContext: null,
     offscreenCanvases: {}, // DOM ID -> OffscreenCanvas mappings of <canvas> elements that have their rendering control transferred to offscreen.
     timerQueriesEXT: [],
-#if USE_WEBGL2
+#if USE_WEBGL2 || USE_WEBGL2_COMPUTE
```

Comment:
Perhaps WEBGL2_COMPUTE should require that WEBGL2 is on, so we don't need to `||` here?

Reply:
I did hesitate for some time if `USE_WEBGL2_COMPUTE` should imply `USE_WEBGL2`. I made the current choice because

* WebGL 2.0 is provided by the `webgl2` context (`WebGL2RenderingContext`) and WebGL 2.0 Compute is provided by the `webgl2-compute` context (`WebGL2ComputeRenderingContext`), so WebGL 2.0 Compute is not simply an extension of WebGL 2.0.
* For a native application in C/C++, the two contexts are usually created with different minor version number.

It gives me the impression that `USE_WEBGL2` and `USE_WEBGL2_COMPUTE` should exist at the same level but be exclusive to each other. Tens of `USE_WEBGL2 || USE_WEBGL2_COMPUTE` is bad, but the other style sounds slightly worse to me at the moment.

Reply:
You could probably introduce another symbol, like HAVE_WEBGL2 or AT_LEAST_WEBGL2, that would be more clear in its intent.

Reply:
As another proposal, use `#if WEBGL_MAJOR_VERSION >= 2`. Not sure if it's better or not.